### PR TITLE
[SYCLomatic] Migrate abs/std::abs to std::abs instead of sycl::abs

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

**Review by intwanghao — APPROVED** [review]
LGTM

**Comment by zhiweij1** [comment]
std::abs works on cuda backend
```
$ cat test.cpp
#include <iostream>
#include <sycl/sycl.hpp>
using namespace sycl;
int main() {
  float data[2];
  data[0] = -1.2;
  queue myQueue;
  device d = myQueue.get_device();
  std::cout << d.get_info<info::device::name>() << std::endl;
  {
    buffer<float, 1> resultBuf{data, range<1>{2}};
    myQueue.submit([&](handler &cgh) {
      accessor writeResult{resultBuf, cgh, read_write};
      cgh.parallel_for<class mykernel>(1, [=](id<1> idx) {
        writeResult[1] = std::abs(writeResult[0]);
      });
    });
  }
  for (int i = 0; i < 2; i++)
    std::cout << "data[" << i << "] = " << data[i] << std::endl;
  return 0;
}
$ SYCL_DEVICE_FILTER="GPU" ./a.out
NVIDIA GeForce GTX 1060 6GB
data[0] = -1.2
data[1] = 1.2
$
```

File: clang/include/clang/DPCT/DPCTOptions.inc [hunk 1]
```diff
@@ -323,6 +323,10 @@ DPCT_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::list<DPCPPExtensions>), NoDPCPPE
              DPCT_OPT_ENUM("enqueued_barriers",
                                      int(DPCPPExtensions::Ext_EnqueueBarrier),
                                      "Enqueued barriers extension.",
+                                     false),
+             DPCT_OPT_ENUM("c_and_cxx_standard_libraries",
```
Comment (zhimingwang36) [inline clang/include/clang/DPCT/DPCTOptions.inc]: Use cxx_standard_library as the extension name.

File: clang/lib/DPCT/AnalysisInfo.h [hunk 2]
```diff
@@ -1934,6 +1934,9 @@ class DpctGlobalInfo {
   static bool useEnqueueBarrier() {
     return getUsingExtension(DPCPPExtensions::Ext_EnqueueBarrier);
```
Comment (zhimingwang36) [inline clang/lib/DPCT/AnalysisInfo.h]: need a test case to see what if cxx_standard_library is off.

Comment (zhiweij1) [reply clang/lib/DPCT/AnalysisInfo.h]: added

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Comment by zhimingwang36** [comment]
Please ask Yury to review the new options.

File: clang/include/clang/DPCT/DPCTOptions.inc [hunk 3]
```diff
@@ -315,21 +315,31 @@ DPCT_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::list<ExplicitNamespace>), UseExp
...
+
+DPCT_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::list<DPCPPExtensionsDefaultDisabled>), UseDPCPPExtensions, "use-dpcpp-extensions",
+           DPCT_OPTION_VALUES(
+             DPCT_OPT_ENUM("c_cxx_standard_library",
+                                     int(DPCPPExtensionsDefaultDisabled::ExtDD_CCXXStandardLibrary),
+                                     "A set of functions from the C and C++ standard libraries which is allowed to\n"
+                                     "be used in SYCL device code.",
+                                     false)
+           ),
+           llvm::cl::desc("Comma separated list of extensions will be used in migrated code.\n"
```
Comment (YuriPlyakhin) [inline clang/include/clang/DPCT/DPCTOptions.inc]:
```suggestion
           llvm::cl::desc("Comma separated list of extensions to be used in migrated code.\n"
```

**Comment by zhiweij1** [comment]
The failed cases is updated in: https://github.com/oneapi-src/SYCLomatic-test/pull/123

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**